robotoff/workers/tasks/__init__.py

```diff
@@ -53,10 +53,38 @@ def delete_product_insights(barcode: str, server_domain: str):
 
 
 EVENT_MAPPING: Dict[str, Callable] = {
+    # 'import_image' is triggered every time there is a new OCR image available for processing by Robotoff, via /api/v1/images/import.
+    #
+    # On each image import, Robotoff performs the following tasks:
+    #  1. Generates various insights based on the OCR-extracted text from the image.
+    #  2. Extracts the nutriscore insight based on the nutriscore ML model.
+    #  3. Triggers the 'object_detection' task, which is described below.
+    #  4. Stores the imported image metadata in the Robotoff DB.
+    #
+    # NOTE: any insights generated by this call are marked for automatic processing, unlike the insights that are generated
```

Comment:
Hum I'm not sure this is true, some insights that are triggered by the webhook call are not automatically processed (such as `product_weight`).

Reply:
Ahh, I was confused by this line:

https://github.com/openfoodfacts/robotoff/blob/837c2da66b80e48a47788dc92d7975bd6c569424/robotoff/workers/tasks/import_image.py#L55


But looking at the code, we only seem to care if this value is `False`:

https://github.com/openfoodfacts/robotoff/blob/837c2da66b80e48a47788dc92d7975bd6c569424/robotoff/insights/importer.py#L230

Reply:
Well the automatic processing logic is a bit cumbersome, basically there are two import mode: automatic or not: this is specified by the `automatic` field in `import_insights` function. If `automatic` is `False`, no imported insight is automatically applied. This is useful when you batch-import insights from CLI (CLI-import is used for instance when we create a new insight type, to import all insights predicted from the JSONL dump).
If `automatic` is `True`, the insight **may** be automatically applied, depending on the return value of `need_validation` function.

Reply:
Thanks for the explanation, added a comment to the cli code :)